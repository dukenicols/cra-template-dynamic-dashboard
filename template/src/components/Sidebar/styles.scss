.sidebar-menu {
    background-color: #4f79a1;
    height: calc(100vh - 56px);
    a {
        color: #ffffff;
    }
    .nav-link {
        &:hover {
            color: #282c34;
        }
    }
}
