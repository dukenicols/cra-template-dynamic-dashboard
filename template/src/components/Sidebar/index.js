import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Nav } from 'react-bootstrap';
import './styles.scss';

function Sidebar({ pages }) {
    return (
        <div className="sidebar-menu">
            <Nav className="flex-column">
                {pages.map((route, index) => {
                    const key = `nav-item-${index}`;
                    return (
                        <Nav.Item key={key}>
                            <Link to={route.path}>
                                {route.name}
                            </Link>
                        </Nav.Item>
                    );
                })}
            </Nav>
        </div>
    );
}

Sidebar.defaultProps = {
    pages: [],
};

Sidebar.propTypes = {
    pages: PropTypes.array,
    onChange: PropTypes.func,
};

export default Sidebar;
