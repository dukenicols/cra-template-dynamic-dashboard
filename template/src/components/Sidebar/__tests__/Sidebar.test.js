import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from '../index';

describe('Sidebar', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Sidebar pages={[]} />);
    });

    it('should render a .sidebar-menu', () => {
        expect(wrapper.exists('.sidebar-menu')).toEqual(true);
    });

    it('should render the menus according to pages prop', () => {
        const testRoutes = [
            {
                name: 'Test 1',
                path: 'test1',
            },
            {
                name: 'Test 2',
                path: 'test2',
            },
        ];
        wrapper.setProps({ pages: testRoutes });
        expect(wrapper.find('NavItem')).toHaveLength(testRoutes.length);
        expect(
            wrapper
                .find('Link')
                .first()
                .text()
        ).toBe(testRoutes[0].name);
        expect(
            wrapper
                .find('Link')
                .last()
                .text()
        ).toBe(testRoutes[testRoutes.length - 1].name);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
