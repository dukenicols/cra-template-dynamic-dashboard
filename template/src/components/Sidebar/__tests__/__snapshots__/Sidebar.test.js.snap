// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match snapshot 1`] = `ShallowWrapper {}`;
